src/OrleansAWSUtils/Membership/DynamoDBMembershipTable.cs

```diff
@@ -0,0 +1,350 @@
+                {
+                    result = false;
+                }
+
+                if (result == false)
+                    logger.Warn(ErrorCode.MembershipBase,
+                        $"Insert failed due to contention on the table. Will retry. Entry {entry.ToFullString()}");
+                return result;
+            }
+            catch (Exception exc)
+            {
+                logger.Warn(ErrorCode.MembershipBase,
+                    $"Intermediate error inserting entry {entry.ToFullString()} to the table {TABLE_NAME_DEFAULT_VALUE}.", exc);
+                throw;
+            }
+        }
+
+        public async Task<bool> UpdateRow(MembershipEntry entry, string etag, TableVersion tableVersion)
+        {
+            try
+            {
+                if (logger.IsVerbose) logger.Verbose("UpdateRow entry = {0}, etag = {1}", entry.ToFullString(), etag);
+                var siloEntry = Convert(entry);
+                int currentEtag = 0;
+                int.TryParse(etag, out currentEtag);
+                siloEntry.ETag = currentEtag;
+                siloEntry.ETag++;
+
+                bool result;
+
+                try
+                {
+                    var conditionalValues = new Dictionary<string, AttributeValue> { { CURRENT_ETAG_ALIAS, new AttributeValue { N = etag } } };
+                    var expression = $"{SiloInstanceRecord.ETAG_PROPERTY_NAME} = {CURRENT_ETAG_ALIAS}";
+                    await storage.UpsertEntryAsync(TABLE_NAME_DEFAULT_VALUE, siloEntry.GetKeys(),
+                        siloEntry.GetFields(), expression, conditionalValues);
+
+                    result = true;
+                }
+                catch (ConditionalCheckFailedException)
+                {
+                    result = false;
+                }
+
+                if (result == false)
+                    logger.Warn(ErrorCode.MembershipBase,
+                        $"Update failed due to contention on the table. Will retry. Entry {entry.ToFullString()}, eTag {etag}");
+                return result;
+            }
+            catch (Exception exc)
+            {
+                logger.Warn(ErrorCode.MembershipBase,
+                    $"Intermediate error updating entry {entry.ToFullString()} to the table {TABLE_NAME_DEFAULT_VALUE}.", exc);
+                throw;
+            }
+        }
+
+        public async Task UpdateIAmAlive(MembershipEntry entry)
+        {
+            try
+            {
+                if (logger.IsVerbose) logger.Verbose("Merge entry = {0}", entry.ToFullString());
+                var siloEntry = ConvertPartial(entry);
+                siloEntry.ETag++;
+                var expression = $"attribute_exists({SiloInstanceRecord.DEPLOYMENT_ID_PROPERTY_NAME}) AND attribute_exists({SiloInstanceRecord.SILO_IDENTITY_PROPERTY_NAME})";
+                await storage.UpsertEntryAsync(TABLE_NAME_DEFAULT_VALUE, siloEntry.GetKeys(), siloEntry.GetFields(), expression);
```

Comment:
Yes you are right. I should don't bump the ETag in that case and only update the IAmAlive field. Just changed. More details on the upsert later on.